src/components/features/Product/ProductOrder/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import { ROUTE_PATH } from '@routes/path';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useAuth } from '@context/auth/useAuth';
+import { Button } from '@components/common';
+import QuantitySelector from './QuantitySelector';
+
+interface ProductOrderProps {
+  name?: string;
+  giftOrderLimit?: number;
+}
+
+export default function ProductOrder({ name, giftOrderLimit }: ProductOrderProps) {
+  const [count, setCount] = useState<number>(0);
+  const { isAuthenticated } = useAuth();
+  const navigate = useNavigate();
+  const { productId } = useParams<{ productId: string }>();
+
+  const handleOrderClick = () => {
+    if (productId) {
+      const orderHistory = { id: Number(productId), count };
+      sessionStorage.setItem('orderHistory', JSON.stringify(orderHistory));
+      const targetPath = isAuthenticated ? ROUTE_PATH.ORDER : ROUTE_PATH.LOGIN;
+      navigate(targetPath);
+    }
+  };
+
+  return (
+    <ProductOrderContainer>
+      <QuantitySelectorConatiner>
+        <Title>{name}</Title>
+        <QuantitySelector giftOrderLimit={giftOrderLimit} onSetCount={setCount} count={count} />
+      </QuantitySelectorConatiner>
+      <div>
+        <TotalAmount>
+          <dl>
+            <dt>총 결제 금액</dt>
+            <dd>145000원</dd>
+          </dl>
+        </TotalAmount>
+        <Button theme="darkGray" onClick={handleOrderClick}>
+          나에게 선물하기
+        </Button>
+      </div>
+    </ProductOrderContainer>
+  );
+}
+
+const ProductOrderContainer = styled.aside`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  max-width: 360px;
+`;
+
+const QuantitySelectorConatiner = styled.div`
```

Comment:
```suggestion
const QuantitySelectorContainer = styled.div`
```

오타 하나 피드백드려요. VSCode 쓰신다면 [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) 추천드려요.